pkg/workloads/spark_job/spark_util.py

```diff
@@ -49,12 +49,34 @@
 }
 
 CORTEX_TYPE_TO_CASTABLE_SPARK_TYPES = {
-    consts.COLUMN_TYPE_INT: [IntegerType(), LongType()],
-    consts.COLUMN_TYPE_INT_LIST: [ArrayType(IntegerType(), True), ArrayType(LongType(), True)],
-    consts.COLUMN_TYPE_FLOAT: [FloatType(), DoubleType(), IntegerType(), LongType()],
-    consts.COLUMN_TYPE_FLOAT_LIST: [ArrayType(FloatType(), True), ArrayType(DoubleType(), True)],
-    consts.COLUMN_TYPE_STRING: [StringType(), IntegerType(), LongType(), FloatType(), DoubleType()],
-    consts.COLUMN_TYPE_STRING_LIST: [ArrayType(StringType(), True)],
+    "csv": {
+        consts.COLUMN_TYPE_INT: [IntegerType(), LongType()],
+        consts.COLUMN_TYPE_INT_LIST: [ArrayType(IntegerType(), True), ArrayType(LongType(), True)],
+        consts.COLUMN_TYPE_FLOAT: [FloatType(), DoubleType(), IntegerType(), LongType()],
+        consts.COLUMN_TYPE_FLOAT_LIST: [
+            ArrayType(FloatType(), True),
+            ArrayType(DoubleType(), True),
+        ],
+        consts.COLUMN_TYPE_STRING: [
+            StringType(),
+            IntegerType(),
+            LongType(),
+            FloatType(),
+            DoubleType(),
+        ],
+        consts.COLUMN_TYPE_STRING_LIST: [ArrayType(StringType(), True)],
```

Comment:
Can you add the array types of the other primitives here? In theory if it could be read, these casts would be supported.